source/adios2/ADIOSTypes.h

```diff
@@ -11,36 +11,169 @@
 #ifndef ADIOS2_ADIOSTYPES_H_
 #define ADIOS2_ADIOSTYPES_H_
 
+/// \cond EXCLUDE_FROM_DOXYGEN
 #include <complex>
 #include <cstddef>
 #include <cstdint>
 #include <limits>
+#include <map>
+#include <string>
 #include <type_traits>
+#include <vector>
+/// \endcond
 
 #include "adios2/ADIOSConfig.h"
 
 namespace adios
 {
 
-// Alias the fixed sized typed into the adios namespace to make sure we're
-// always using the right ones.
+/** Variable shape type identifier */
+enum class ShapeID
+{
+    GlobalValue, ///< single global value, common case
+    GlobalArray, ///< global (across MPI_Comm) array, common case
+    JoinedArray, ///< global array with a common (joinable) dimension
+    LocalValue,  ///< special case, local independent value
+    LocalArray   ///< special case, local independent array
+};
+
+/** Used to set IO class */
+enum class IOMode
+{
+    Independent, ///< all I/O operations are independent per rank
+    Collective   ///< expect collective I/O operations
+};
+
+enum class OpenMode
+{
+    Undefined,
+    Write,
+    Read,
+    Append,
+    ReadWrite,
+    w,
+    r,
+    a,
+    rw
+};
+
+typedef enum {
+    GLOBAL_READERS = 2,
+    ROUNDROBIN_READERS = 3,
+    FIFO_READERS = 4,
+    OPEN_ALL_STEPS = 5
+} ReadMultiplexPattern;
+
+typedef enum {
+    NOWAITFORSTREAM = 0,
+    WAITFORSTREAM = 1
+} StreamOpenMode; // default: wait for stream
+
+enum class TransformType
+{
+    bzip2,
```

Comment:
Certainly.  I'm looking at implementing some clang-tidy checks that can enforce variable naming for members, data members, classes, enums, local variables, etc., but the config file is really confusing for that particular rule.  Until then, we'll catch it via code review.